src/main/java/mods/railcraft/common/blocks/tracks/TrackSwitchBase.java

```diff
@@ -102,33 +106,30 @@ protected boolean isSwitched(EntityMinecart cart) {
             return false;
         }
 
-        // we're not sprung or locked so we should return switched
-        return switched;
+        // we're not sprung or locked so we should return shouldSwitch
+        return shouldSwitch;
     }
 
     public boolean isLocked() {
         return locked > 0;
     }
 
-    //    @Override
-    //    public boolean blockActivated(EntityPlayer player)
-    //    {
-    //        ItemStack current = player.getCurrentEquippedItem();
-    //        if(player.isSneaking() && current != null && current.getItem() instanceof ICrowbar) {
-    //            int meta = tileEntity.getBlockMetadata();
-    //            getWorld().setBlockMetadata(getX(), getY(), getZ(), meta == 0 ? 1 : 0);
-    //            markBlockNeedsUpdate();
-    //            if(current.isItemStackDamageable()) {
-    //                current.damageItem(1, player);
-    //            }
-    //            return true;
-    //        }
-    //        return super.blockActivated(player);
-    //    }
     @Override
     public void onBlockPlaced() {
         determineTrackMeta();
         determineMirror();
+
+        // Notify any neighboring switches that we exist so they know to register themselves with us
+        Block myBlock = getWorld().getBlock(getX(), getY(), getZ());
+        getWorld().notifyBlocksOfNeighborChange(getX(), getY(), getZ(), myBlock);
```

Comment:
there should be a shortcut function for this on the super class notifyBlocksOfNeighborChange().